src/transport_ofi.h

```diff
@@ -204,78 +234,105 @@ typedef struct shmem_transport_ofi_bounce_buffer_t shmem_transport_ofi_bounce_bu
 
 typedef int shmem_transport_ct_t;
 
-extern shmem_free_list_t *shmem_transport_ofi_bounce_buffers;
+struct shmem_transport_ctx_t {
+  int                             id;
+#ifdef USE_CTX_LOCK
+  shmem_internal_mutex_t          lock;
+#endif
+  long                            options;
+  struct fid_ep*                  cntr_ep;
+  struct fid_ep*                  cq_ep;
+  struct fid_stx*                 stx;
+  struct fid_cntr*                put_cntr;
+  struct fid_cntr*                get_cntr;
+  struct fid_cq*                  cq;
+  shmem_internal_atomic_uint64_t  pending_put_cntr;
+  shmem_internal_atomic_uint64_t  pending_get_cntr;
+  shmem_free_list_t              *bounce_buffers;
+};
+
+typedef struct shmem_transport_ctx_t shmem_transport_ctx_t;
+extern shmem_transport_ctx_t shmem_transport_ctx_default;
+
+extern struct fid_ep* shmem_transport_ofi_target_ep;
+
+#ifdef USE_CTX_LOCK
+#define SHMEM_TRANSPORT_OFI_CTX_LOCK(ctx)                               \
+    if (!((ctx)->options & (SHMEM_CTX_PRIVATE | SHMEM_CTX_SERIALIZED))) \
+        SHMEM_MUTEX_LOCK((ctx)->lock);
+#define SHMEM_TRANSPORT_OFI_CTX_UNLOCK(ctx)                             \
+    if (!((ctx)->options & (SHMEM_CTX_PRIVATE | SHMEM_CTX_SERIALIZED))) \
+         SHMEM_MUTEX_UNLOCK((ctx)->lock);
```

Comment:
Should we check somewhere that --enable-thread-completion isn't being combined with --disable-threads?

Reply:
Providing both options causes the build to fail..  Fixing this at the configury level.